hw/ip_templates/otp_ctrl/dv/env/seq_lib/otp_ctrl_zeroization_vseq.sv.tpl

```diff
@@ -0,0 +1,276 @@
+        // be return the current value of the zeroized otp location
+        csr_rd(ral.direct_access_rdata[0], rdata0);
+        if (is_granule_64(dai_addr))
+          csr_rd(ral.direct_access_rdata[1], rdata1);
+
+        `uvm_info(`gfn, $sformatf("ZRing(Post): part_idx :%s, dai_addr :%x, rdata0:%x, rdata1:%x",
+                  part_idx.name, dai_addr, rdata0, rdata1), UVM_LOW);
+      end
+    end
+
+    `uvm_info(`gfn, $sformatf("Starting Clock Delay - 1"), UVM_LOW);
+    // Wait a few clocks before issuing a readback to check if the address is zeroized and holds
+    // state post reset
+    cfg.clk_rst_vif.wait_clks(10);
+    `uvm_info(`gfn, $sformatf("Done Clock Delay - 1"), UVM_LOW);
+
+    `uvm_info(`gfn, "Reseting OTP - Zeroization State should be saved", UVM_LOW);
+    do_otp_ctrl_init = 0;
+    dut_init();
+    dut_init_completed = 1;
+
+    // dut_init() also sets interrupt enables. For this sequence otp_operation_done is disabled as
+    // we are expliciting testing otp_error.
+
+    // However the prediction model for otp_operation_done
+    // does need a revisit to make sure the interrupt state registers are set at the time the RTL
+    // also sets the same register and not the current way the model is coded
+
+
+    if (used_zeroized_addrs.num() != 0) begin
+      `uvm_info(`gfn, $sformatf("used_zeroized_addrs[]: %d", used_zeroized_addrs.num()), UVM_LOW);
+    end
+
+    foreach (used_zeroized_addrs[addr]) begin
+      `uvm_info(`gfn, $sformatf("Reading back ZRed Addr: %x", addr), UVM_LOW);
+      dai_rd(.addr(addr), .rdata0(rdata0), .rdata1(rdata1), .skip_idle_check(1));
+    end
+
+    `uvm_info(`gfn, $sformatf("Delay before sequence termination"), UVM_LOW);
+    cfg.clk_rst_vif.wait_clks(25);
+    `uvm_info(`gfn, $sformatf("Done Delay"), UVM_LOW);
+  endtask : body
+
+  // This is an interrupt service routine that waits for interrupt and clears the following
+  // Registers
+  // - INTR
+  // - STATUS
+  // - ERROR CODE
+  task intr_service_routine();
+    bit [DataWidth-1:0] intr_enable;
+    bit [DataWidth-1:0] csr_rd_value;
+
+    `uvm_info(`gfn, $sformatf("intr_service_routine - Starting"), UVM_LOW);
+    wait (cfg.under_reset == 0);
+
+    `uvm_info(`gfn, $sformatf("(ISR) Starting foreverloop for Reset / Interrupt"), UVM_LOW);
+
+    forever begin
+      @ (posedge cfg.under_reset or posedge cfg.intr_vif.pins[1]) begin
+        if (cfg.under_reset) begin
+          `uvm_info(`gfn, $sformatf("(ISR)- Reset Seen"), UVM_LOW);
+          // Reset has been triggered, wait until OTP is initialized
+          wait (cfg.under_reset == 0);
+          wait (cfg.otp_ctrl_vif.pwr_otp_done_o == 1'b1);
+          wait (dut_init_completed == 1);
+
+          `uvm_info(`gfn, $sformatf("(ISR) DUT Init Complete-Staring wait for Interrrupt"), UVM_LOW);
+          dut_init_completed = 0;
+        end else if (cfg.intr_vif.pins[1] == 1'b1) begin
```

Comment:
I thought you had a TODO in mind for the other interrupt? This could be a good place to mention it.

By the way, can this isr be used for most other sequences?